actors/miner/src/monies.rs

```diff
@@ -173,41 +185,37 @@ pub fn pledge_penalty_for_termination_lower_bound(
     )
 }
 
-/// Penalty to locked pledge collateral for the termination of a sector before scheduled expiry.
-/// SectorAge is the time between the sector's activation and termination.
-#[allow(clippy::too_many_arguments)]
+// TODO: write better description, mention FIP-0098
 pub fn pledge_penalty_for_termination(
-    day_reward: &TokenAmount,
+    initial_pledge: &TokenAmount,
     sector_age: ChainEpoch,
-    twenty_day_reward_at_activation: &TokenAmount,
-    network_qa_power_estimate: &FilterEstimate,
-    qa_sector_power: &StoragePower,
-    reward_estimate: &FilterEstimate,
-    replaced_day_reward: &TokenAmount,
-    replaced_sector_age: ChainEpoch,
+    fault_fee: &TokenAmount,
 ) -> TokenAmount {
-    // max(SP(t), BR(StartEpoch, 20d) + BR(StartEpoch, 1d) * terminationRewardFactor * min(SectorAgeInDays, 140))
-    // and sectorAgeInDays = sectorAge / EpochsInDay
-    let lifetime_cap = TERMINATION_LIFETIME_CAP * EPOCHS_IN_DAY;
-    let capped_sector_age = std::cmp::min(sector_age, lifetime_cap);
-
-    let mut expected_reward: TokenAmount = day_reward * capped_sector_age;
-
-    let relevant_replaced_age =
-        std::cmp::min(replaced_sector_age, lifetime_cap - capped_sector_age);
-
-    expected_reward += replaced_day_reward * relevant_replaced_age;
-
-    let penalized_reward = expected_reward * TERMINATION_REWARD_FACTOR_NUM;
-    let penalized_reward = penalized_reward.div_floor(TERMINATION_REWARD_FACTOR_DENOM);
+    // Use the _percentage of the initial pledge_ strategy to determine the termination fee.
+    let termination_fee =
+        (initial_pledge * TERM_PENALTY_PLEDGE_NUM).div_floor(TERM_PENALTY_PLEDGE_DENOM);
+
+    // There are a few special cases to consider where the termination fee must be tweaked to
+    // maintain the current network conditions.
+
+    // 1. We need to ensure that the final termination fee is always greater than the fault fee for
+    //    the same sector.
+    let fault_fee = (fault_fee * FAULT_FEE_MULTIPLE_NUM).div_floor(FAULT_FEE_MULTIPLE_DENOM);
+    let termination_fault_max = cmp::max(termination_fee, fault_fee);
+
+    // 2. We need to ensure linear growth of the termination fee over time, up to a cap. The
+    //    details of this growth are defined in FIP-0098 design rationale.
+    let sector_age_in_days = sector_age / EPOCHS_IN_DAY;
+    let penalty_ramped = cmp::min(
+        (&termination_fault_max * sector_age_in_days).div_floor(TERMINATION_LIFETIME_CAP),
+        termination_fault_max,
+    );
```

Comment:
we're going to hit a sector age of `0` here if it's young enough I think, maybe we should move the days denominator down:

```suggestion
    let penalty_ramped = cmp::min(
        (&termination_fault_max * sector_age).div_floor(TERMINATION_LIFETIME_CAP * EPOCHS_IN_DAY),
        termination_fault_max,
    );
```

Reply:
I think also this shouldn't be applied at all for sectors older than `TERMINATION_LIFETIME_CAP`, and it probably shouldn't be multiplying `termination_fault_max` since it's a fixed max cap, so maybe the fault_fee calculation could be moved down and then you get the 2%pledge as a minimum and use `cmp::clamp()` to bound the 140-days scaled version

Reply:
To be clear, do you mean something along those lines?

```rust
    let termination_fee =
        (initial_pledge * TERM_FEE_PLEDGE_MULTIPLE_NUM).div_floor(TERM_FEE_PLEDGE_MULTIPLE_DENOM);

    let termination_fee_age_adjusted = cmp::min(
        termination_fee.clone(),
        (sector_age * &termination_fee).div_floor(TERMINATION_LIFETIME_CAP * EPOCHS_IN_DAY),
    );

    let minimum_fee = (initial_pledge * TERM_FEE_MIN_PLEDGE_MULTIPLE_NUM)
        .div_floor(TERM_FEE_MIN_PLEDGE_MULTIPLE_DENOM);

    let maximum_fee = (fault_fee * TERM_FEE_MAX_FAULT_FEE_MULTIPLE_NUM)
        .div_floor(TERM_FEE_MAX_FAULT_FEE_MULTIPLE_DENOM);
    termination_fee_age_adjusted.clamp(minimum_fee, maximum_fee).clone()
```

If so, can there be a case where `minimum_fee` > `max_fee`? This would panic the `clamp`, so probably some measure of defense mechanisms should be put in place.

Reply:
apparently this is possible, this should go back to the FIP for clarification

Reply:
New spec from https://github.com/filecoin-project/FIPs/pull/1128/commits/a31032e3fc1a070dc7ce279e9098f1bf09c153e2 and implemented in https://github.com/filecoin-project/builtin-actors/pull/1639/commits/c1124049abdfe42418d23450782aa7d47410434a so it doesn't seem relevant anymore.